web/src/message_viewport.ts

```diff
@@ -58,7 +82,11 @@ export function message_viewport_info() {
     // generally be concerned about whether actual message content is
     // visible.
 
-    const res = {};
+    const res: MessageViewportInfo = {
+        visible_top: 0,
+        visible_bottom: 0,
+        visible_height: 0,
+    };
```

Comment:
A cleaner approach might be to just set local variables using `let visible_top = ...` and then define this as `const res: MessageViewportInfo = {visible_top, visible_bottom, visible_height}` at the bottom.

Reply:
Agreed, refactored this.